src/VirtualModulePlugin.ts

```diff
@@ -167,10 +167,14 @@ class VirtualModulesPlugin {
           : finalWatchFileSystem.watcher.fileWatchers;
 
       fileWatchers.forEach((fileWatcher) => {
-        if (fileWatcher.path === filePath) {
+        const fileWatcherPath = fileWatcher.path || fileWatcher.watcher.path;
```

Comment:
What about using 
```
fileWatcher = fileWatcher.watcher ?? fileWatcher
```
This way we need a single change, would this work?

Reply:
Checked on a local dependency works for me

Reply:
It will work, yes, it's just so that it feels wrong to overwrite argument, can make this change if need be anyway. Other thing is that I actually have no idea if old API had no `watcher` property in this object.